### PR TITLE
enhancement: improve tracking for "unable to load"

### PR DESCRIPTION
**Pull request by MounirDhahri**
The type of this PR is: **TYPE**

This PR resolves [CX-1703]

### Description

This PR contains 2 parts that you can navigate by commit:
1. **Leverage Datadog/Statd to track graphql request failures**
   i. Include tracking as part of the relay errorMiddleware
  ii. Include the **operation kind** (mutation/query), **query name** (HomeQuery...) and the **handler** (default, @optionalField or @principalField). The handler is needed here to know exactly how much the @optionalField is helping us.
2. **Remove old way of tracking those failures**

_The board looks like this on Datadog_** 
Screenshot 2021-08-11 at 17 47 06

Link: https://app.datadoghq.com/dashboard/ib7-8gr-u6x/eigen-graphql-requests-with-errors

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- improve tracking for "unable to load" - mounir

</details>

[CX-1703]: https://artsyproduct.atlassian.net/browse/CX-1703

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (improve tracking for "unable to load" - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 839c369950a2266b030c14dd7910107b746e504d
</p>

**Review by olerichter00 — APPROVED** [review]
LGTM 🥳

File: src/lib/relay/middlewares/errorMiddleware.ts [hunk 1]
```diff
@@ -43,6 +44,11 @@ export const errorMiddleware = () => {
     const allErrorsAreOptional = resJson.extensions?.optionalFields?.length === resJson.errors?.length
 
     if (allErrorsAreOptional) {
+      volleyClient.send({
+        type: "increment",
+        name: "graphql-request-with-errors",
+        tags: [`query:${req.operation.name}`, `kind:${req.operation.kind}`, "handler: optionalField"],
+      })
```
Comment (olerichter00) [inline src/lib/relay/middlewares/errorMiddleware.ts:51]: **(idea):** Shall we move this into a method? Something like `trackError(handler: "optionalField" | "default" | "principalField")`?

Comment (MounirDhahri) [reply src/lib/relay/middlewares/errorMiddleware.ts:51]: yes, that's a good idea. I will do it

**Comment by MounirDhahri** [comment]
#squashongreen

**Pull request merged**